libtenzir/src/http_api.cpp

```diff
@@ -225,6 +225,23 @@ auto parse_endpoint_parameters(const tenzir::rest_endpoint& endpoint,
           }
           return result;
         },
+        [&](const record_type&) -> caf::expected<data> {
+          const auto* parsed = caf::get_if<record>(&param_data);
+          if (!parsed)
+            return caf::make_error(ec::invalid_argument, "expected a record");
+          auto result = record{};
+          for (const auto& x : *parsed) {
+            auto parse_result = false;
+            if (!parsers::boolean(caf::get<std::string>(x.second),
+                                  parse_result)) {
+              return caf::make_error(ec::invalid_argument,
+                                     "currently only boolean values in records "
+                                     "are supported");
+            }
+            result[x.first] = parse_result;
+          }
```

Comment:
I was surprised to see that we get a string here instead of a bool, but we seems that we pass bools as their raw string representation for these endpoints...

However, you want `caf::get_if<std::string>` here, as it can still be something else besides `std::string` (and `caf::get` on `caf::variant` with the wrong type is sadly UB...)

Reply:
Yep, a segfault can be triggered by passing `{{key : null}}` as an HTTP endpoint parameter. I made use of `caf::get_if` in https://github.com/tenzir/tenzir/pull/3585/commits/2c75d49eae971c37db2f883e0384567d5b0f5ec5